rpcs3/Emu/Io/usb_device.cpp

```diff
@@ -200,75 +200,64 @@ bool usb_device_emulated::open_device()
 
 u32 usb_device_emulated::get_descriptor(u8 type, u8 index, u8* buf, u32 buf_size)
 {
-	u32 expected_count = 2;
+	const std::array<u8, 2> header {header.size(), type};
+	u32 expected_count = buf ? std::min<u32>(header.size(), buf_size) : 0;
+	memcpy(buf, header.data(), expected_count);
 
-	if (buf_size < expected_count)
+	if (expected_count >= header.size())
```

Comment:
Avoid changing so many lines, use an early return when this evaluates to false.

Reply:
Okay. Done.